### DISCUSSION TITLE
How to clear GQL cache on specific queries.

### DISCUSSION THREAD

**Discussion by aodihis** [craftcms/cms#13159] [Q&A]
I have a custom query for graphQL and also I enable the gql cache.
I wanted on specific period of time to clear all cache for that custom query without clearing the whole gql cache.

Is it something possible to achieve?

**Comment by brandonkelly** [comment]
It’s not currently possible to invalidate a single query’s result cache. Query caches do automatically get invalidated whenever a referenced element gets updated, though. Is that good enough, or is there some reason you need direct control?

**Comment by aodihis** [comment]
I see, actually yes, because I have a gql query to show the recommended entries. So basically I have a custom table that store the entries and the recommended entries id, and every 2 hours it will be regenerated.

But anyway thanks for the answer.
